### PR TITLE
Upload / delete automation tests

### PR DESCRIPTION
**Pull request by PoroQa**

### CODE REVIEW

File: QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        given().relaxedHTTPSValidation()
+                .header("Authorization","Bearer " + token)
+        .when()
+                .delete(Endpoints.DELETE_FILE + Endpoints.DELETE_FILE_PATH + fileId)
+        .then()
+                .statusCode(200);
+    }
+
+    public static String getFileIdFromFileName(String email, String pass, int fileIndexPosition) {
```
Comment (PoroQa) [inline QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java]: Method name will be changed to "getFileIdByJsonIndex" or something like that. I don't think it's a good solution to delete files like this but couldn't think of any another solution.

File: QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            throw new RuntimeException(e);
+        }
+    }
+
+    //This method upload number of files that are located in "filePath.properties" file, located in TestResources
+    public static void uploadThenDeleteFiles(int numOfFiles) {
+        String token = UserDetailsGenerator.takeTokenValueFromJson(fileLogin.getValue("VALID_EMAIL"), fileLogin.getValue( "VALID_PASS_CREATED_USER"));
+
+        for (int i = 0; i <= numOfFiles; i++) {
+            given().relaxedHTTPSValidation()
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java]: think that this code block is duplicate and already extracted in uploadFile method

File: QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    .multiPart(new File(filePath.getValue("FILEPATH" + i)))
+            .when()
+                    .post(Endpoints.UPLOAD_FILE)
+            .then()
+                    .statusCode(200);
+
+            fileIds.add(UserDetailsGenerator.takeValueFromResponseWithToken(Endpoints.READ_FILE, token,i,"fileId"));
+        }
+        for (int i = 0; i <= numOfFiles; i++) {
+            given().relaxedHTTPSValidation()
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java]: This code block is already extracted in deleteFile method

File: QA/PoroDevTest/src/main/java/common/api_setup/api_common/UserDetailsGenerator.java [hunk 4]
```diff
@@ -43,36 +43,36 @@ public static String convertToJson(Object object) {
...
 
-        return fetchedToken;
+        return JsonPath.read(json, "$.jwt").toString();
     }
 
     // Generating JSON body for runtime with two numbers
     public static String createRuntimeJsonReq(String fileID, String firstNmb, String secondNmb) {
-        String generatedRuntimeJsonReq = "{\n" +
-                "  \"projectId\": \""+fileID+"\",\n" +
+        return "{\n" +
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/src/main/java/common/api_setup/api_common/UserDetailsGenerator.java:64]: for which request this requestBody is called?

File: QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    private final FileControlUtil fileLogin = new FileControlUtil(FileControlUtil.LOGIN_DATA_PROPERTIES);
+    private final FileControlUtil filePath = new FileControlUtil(FileControlUtil.FILE_PATH_PROPERTIES);
+    private final Logger logger = LoggerFactory.getLogger(UploadAndDeleteFile.class);
+
+    public UploadAndDeleteFile() throws IOException {
+    }
+
+    @Test(priority = 1)
+    public void upload_xml_file() {
+        DataControlUtil.uploadFile(
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java]: Please add some assertion method in order to confirm that test is successfull

File: QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                fileLogin.getValue("VALID_EMAIL"),
+                fileLogin.getValue("VALID_PASS_CREATED_USER"),
+                filePath.getValue("POM_FILE"));
+        logger.info("Xml file is uploaded, status code is 200");
+    }
+
+    @Test(priority = 2)
+    public void delete_xml_file() {
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java]: Add some assertions, beside statusCode check
Each of assertion can be stored directly in test method
Please provide an additional assertions in rest of test-cases

File: QA/PoroDevTest/src/test/resources/filePath.properties [hunk 7]
```diff
@@ -0,0 +1,25 @@
+FILEPATH0 = C:\\Users\\User\\Documents\\Lightshot\\Screenshot_7.png
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/src/test/resources/filePath.properties:1]: most of these files can be part of a project, stored in java/test/resources dir

File: QA/PoroDevTest/testng.xml [hunk 8]
```diff
@@ -1,14 +1,14 @@
 <!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
 
-<suite name="Suite1" verbose="1" >
-    <test name="Login Tests" >
+<suite name="Suite1" verbose="1">
```
Comment (udrakulic-htec) [inline QA/PoroDevTest/testng.xml]: Please add more meaningful suite name

**Review by udrakulic-htec — CHANGES_REQUESTED** [review]

File: QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java (hunk 2 above)
Comment (PoroQa) [reply QA/PoroDevTest/src/main/java/common/api_setup/api_common/DataControlUtil.java]: Good point, will be resolved

File: QA/PoroDevTest/src/main/java/common/api_setup/api_common/UserDetailsGenerator.java (hunk 4 above)
Comment (PoroQa) [reply QA/PoroDevTest/src/main/java/common/api_setup/api_common/UserDetailsGenerator.java:64]: To execute a Runtime project that takes two numbers as a parameter. This runtime project adds these two numbers

File: QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java [hunk 9]
```diff
@@ -17,93 +20,50 @@ public class UploadAndDeleteFile extends ApiConfig {
...
-    }
-
-    @Test(priority = 8)
-    public void upload_500mb_file() {
-        DataControlUtil.uploadFile(
-                fileLogin.getValue("VALID_EMAIL"),
-                fileLogin.getValue("VALID_PASS_CREATED_USER"),
-                filePath.getValue("FIVE_HUNDRED_MB_FILE"));
-        logger.info("Five hundred mb file is uploaded, status code is 200");
+        logger.info("File is not deleted, still exists in the database, status code is 200");
```
Comment (PoroQa) [inline QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java:61]: Will be changed

Comment (udrakulic-htec) [inline QA/PoroDevTest/src/test/java/api_test/files/UploadAndDeleteFile.java:61]: assertions still miss

**Review by udrakulic-htec — APPROVED** [review]

**Pull request merged**